ordersCostumes.js

```diff
@@ -0,0 +1,117 @@
+const addCostumeToOrder = async (pool, costumeId, orderId) => {
+    // throw error if costume id does not exist
+    const {rows:[costume]} = await pool.query(`
+        SELECT * FROM costumes
+        WHERE id = $1;
+    `, [costumeId])
+    if (costume === undefined) {
+        throw new Error(`Could not retrieve data because id provided for costume (${id}) does not exist in table.`)
+    } 
+
+    // throw error if order id does not exist
+    const {rows:[order]} = await pool.query(`
+        SELECT * FROM orders
+        WHERE id = $1;
+    `, [orderId])
+    if (order === undefined) {
+        throw new Error(`Could not retrieve data because id provided for order (${id}) does not exist in table.`)
+    } 
+    
+    // add costume
+    const {rows: [costumeOrder]} = await pool.query(`
+        INSERT INTO orders_costumes(
+            costume_id,
+            order_id
+        ) VALUES ($1,$2)
+        RETURNING *;
+    `, [costumeId, orderId]
+    )
+    return costumeOrder;
+}
+
+const removeCostumeFromOrder = async (pool, costumeId, orderId) => {
+    // throw error if costume id does not exist
+    const {rows:[costume]} = await pool.query(`
+        SELECT * FROM costumes
+        WHERE id = $1;
+    `, [costumeId])
+    if (costume === undefined) {
+        throw new Error(`Could not retrieve data because id provided for costume (${id}) does not exist in table.`)
+    } 
+
+    // throw error if order id does not exist
+    const {rows:[order]} = await pool.query(`
+        SELECT * FROM orders
+        WHERE id = $1;
+    `, [orderId])
+    if (order === undefined) {
+        throw new Error(`Could not retrieve data because id provided for order (${id}) does not exist in table.`)
+    } 
+    
+    // remove costume
+    await pool.query(`
+        DELETE FROM orders_costumes
+        WHERE costume_id = $1 AND order_id = $2;
+    `, [costumeId, orderId]);
+}
+
+const getAllOrdersOfCostumeById = async (pool, costumeId) => {
```

Comment:
I'm not sure if this is needed? What is the feature you're thinking of that would require this?
```suggestion
const getOrdersContainingCostumeId = async (pool, costumeId) => {
```

Reply:
Thought it might be useful for costume shop business's analytics. For example, you could see what other costumes were in an order that had a specific costume in it, and extrapolate that "people who buy _x_ also buy _y_". Perhaps a feature that is not essential at this juncture.

Reply:
I was trying to come up with more adapters the might be useful in hypothetical scenarios versus ones that would have a function right now in the project as is.

Reply:
That makes sense to me! Keep it, maybe add a comment about possible use-case so you can remember later